manager/base/parsers.go

```diff
@@ -0,0 +1,180 @@
+package base
+
+import (
+	"app/base/models"
+	"errors"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// ParseBoolArray, parses bool array in query arguments,
+// array can be checked for max values by limit
+func ParseBoolArray(rawValues []string, limit *int) ([]bool, error) {
+	if limit != nil && len(rawValues) != *limit {
+		return []bool{}, errors.New("Invalid bool array format")
+	}
+	var res []bool
+	for _, rawVal := range rawValues {
+		val, err := strconv.ParseBool(rawVal)
+		if err != nil {
+			return res, errors.New("Invalid bool value in bool array")
+		}
+		res = append(res, val)
+	}
+	return res, nil
+}
+
+// ParseDateRange, parses 2 member array with date range
+func ParseDateRange(rawValues []string) ([]time.Time, error) {
+	if len(rawValues) != 2 {
+		return []time.Time{}, errors.New("Invalid date range format")
+	}
+	dateFrom, err := time.Parse(DateFormat, rawValues[0])
+	if err != nil {
+		return []time.Time{}, errors.New("Invalid date format")
+	}
+	dateTo, err := time.Parse(DateFormat, rawValues[1])
+	if err != nil {
+		return []time.Time{}, errors.New("Invalid date format")
+	}
+	return []time.Time{dateFrom, dateTo}, nil
+}
+
+// ParseSeverity, parses array of severity strings
+func ParseSeverity(rawValues []string) ([]models.Severity, error) {
+	var res []models.Severity
+	for _, raw := range rawValues {
+		raw = strings.ToLower(raw)
+		switch raw {
+		case "", "null":
+			res = append(res, models.NotSet)
+		case "none":
+			res = append(res, models.None)
+		case "low":
+			res = append(res, models.Low)
+		case "medium":
+			res = append(res, models.Medium)
+		case "moderate":
+			res = append(res, models.Moderate)
+		case "important":
+			res = append(res, models.Important)
+		case "high":
+			res = append(res, models.High)
+		case "critical":
+			res = append(res, models.Critical)
+		default:
+			return res, errors.New("Invalid severity argument")
+		}
+	}
+	return res, nil
+}
+
+// ParseCvssScoreRange parses array of two member range of cvss score floats
+func ParseCvssScoreRange(rawValues []string) ([]float32, error) {
+	if len(rawValues) != 2 {
+		return []float32{}, errors.New("Invalid cvss_score range format")
+	}
+	scoreFrom, err := strconv.ParseFloat(rawValues[0], 32)
+	if err != nil {
+		return []float32{}, errors.New("Invalid cvss score from value")
+	}
+	scoreTo, err := strconv.ParseFloat(rawValues[1], 32)
+	if err != nil {
+		return []float32{}, errors.New("Invalid cvss score to value")
+	}
+	return []float32{float32(scoreFrom), float32(scoreTo)}, nil
+}
+
+// ParseInt parses string to int64
+func ParseUint(rawValues []string) (uint64, error) {
```

Comment:
Why the list as an input param?

Reply:
when you put HTTP param as `param=1,2,3` the gin parses it as `[]string{"1", "2", "3"}` , this parsers functions are called from param (filter) parser function and checks if it is valid uint as param. (In this code the first check if the length is 1). I know this http param length could be checked in the big switch case but I think it would get really big single function.